confgenerator/fluentbit/modular.go

```diff
@@ -43,14 +43,29 @@ func (c Component) generateSection() string {
 	return fmt.Sprintf("[%s]\n%s\n", c.Kind, strings.Join(lines, "\n"))
 }
 
-type Pipeline struct {
+type ModularConfig struct {
+	Variables  map[string]string
 	Components []Component
-	LogName    string
-	Internal   bool
 }
 
-type ModularConfig struct {
-	Pipelines      []Pipeline
-	InternalOutput Component
-	Output         Component
+func (c ModularConfig) Generate() (string, string, error) {
```

Comment:
Would help to name the output parameters…

Reply:
That's not what named output parameters are for. Go guidance is that you should not name output parameters unless you specifically need them.

Reply:
I didn't find any explicit guidance in either the Go style or the Go best practices on the named parameters. However, Go is supposed to be all about not surprising readers, and not specifying which of the strings is which opens up a possibility of mixing them up. Would you mind pointing me at the guidance about output parameters that you're referencing?

Reply:
The relevant style entry is https://golang.org/doc/effective_go#named-results (plus more internal guidance that I can't link to in a public PR). I agree that's not a strong recommendation; I have personally been given the explicit feedback in readability reviews that named result parameters should not be used unless those names are needed inside the function.

This function's return parameters will obviously be documented with a comment.

Reply:
My understanding from going through the readability process is that naming result parameters makes it possible to use a bare `return`, and if you're not actually using that in the function body, you should not use named result parameters to prevent an accidental bare return.

I agree this seems like a fairly weak argument. I think I'm fine going against that guidance if you really want. (Note that this function signature is not new and it didn't have named result parameters before...)